ixblue_c3_ins/include/ixblue_c3_ins/c3_protocol.h

```diff
@@ -0,0 +1,78 @@
+/*  QinetiQ North America (QNA)
+ *  350 Second Avenue
+ *  Waltham, MA 02451
+ *
+ *  Proprietary Licensed Information.
+ *  Not to be duplicated, used, or disclosed,
+ *  except under terms of the license.
+ *
+ *  Copyright © 2018 QinetiQ North America  All rights reserved.
+ */
+
+#ifndef IXBLUE_C3_INS_C3_PROTOCOL_H
+#define IXBLUE_C3_INS_C3_PROTOCOL_H
+
+#include <stdint.h>
+#include <sys/types.h>
+
+namespace ixblue_c3_ins
+{
+namespace c3_protocol
+{
+#define NAV_LONG_MSG_LENGTH 90
+/* contents of a ixBlue C3 'navigation long' packetconverted
+ * into a host data structure for publishing to other nodes.
+ */
+class nav_long
+{
+public:
+  // constructor/destructor
+  nav_long(char *buf, ssize_t nbytes);
+  ~nav_long();
+  // accessors
+
+  struct nav_long_data_t
```

Comment:
I see different data types between nav_long_data_t struct and NavigationLong.msg eg:
- int32_t latitude / float64 latitude

Reply:
That's right ! The INS sends latitude and longitude information as signed 32 bit integers with a quantization constant (check NAVIGATION LONG packet in the [INS manual](https://www.igp.de/manuals/7-INS-InterfaceLibrary_MU-INSIII-AN-001-O.pdf)), but we publish them as floating point quantities after applying the corresponding constant.

Reply:
Ok